src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -110,73 +117,97 @@ void Run (DirectoryAssemblyResolver res)
 			}
 
 			// Put every assembly we'll need in the resolver
+			bool hasExportReference = false;
+			bool haveMonoAndroid = false;
```

Comment:
Do we really care about/need `haveMonoAndroid` and `hasExportReference`?  "Worst" case, we'll try to add these assemblies to the `HashSet<string>`, which I would *hope* wouldn't actually take much time.  Does this really save anything?

Reply:
Yes, it saves a `String.Compare` call for each assembly in `Resolved.Assemblies`. With `Mono.Android` being usually at the top of the array, it does save cycles.